src/include/storage/tuple_access_strategy.h

```diff
@@ -253,6 +253,40 @@ class TupleAccessStrategy {
    */
   const BlockLayout &GetBlockLayout() const { return layout_; }
 
+  /**
+   * Compare and swap block status to busy. Expect that block to be idle, if yes, set the block status to be busy
+   * @param block the block to compare and set
+   * @return true if the set operation succeeded, false if the block is already busy
+   */
+  bool SetBlockBusyStatus(RawBlock *block) const {
+    uint32_t old_val = ClrBit(block->insert_head_.load());
+    return block->insert_head_.compare_exchange_weak(old_val, SetBit(old_val));
+  }
+
+  /**
+   * Compare and swap block status to idle. Expect that block to be busy, if yes, set the block status to be idle
+   * @param block the block to compare and set
+   * @return true if the set operation succeeded, false if the block is already idle
+   */
+  bool ClearBlockBusyStatus(RawBlock *block) const {
+    uint32_t old_val = SetBit(block->insert_head_.load());
+    return block->insert_head_.compare_exchange_strong(old_val, ClrBit(old_val));
+  }
+
+  /**
+   * Set the first bit of given val to 0, helper function used by ClearBlockBusyStatus
+   * @param val the value to be set
+   * @return the changed value with first bit set to 0
+   */
+  static uint32_t ClrBit(uint32_t val) { return val & ~(1 << 31); }
```

Comment:
nit: don't use abbreviations here for clr as other parts of your code are not using them